crates/threads-xform/tests/basic.wat

```diff
@@ -0,0 +1,150 @@
+;; @xform export "foo" (externref_borrowed)
+
+(module
+  (import "env" "memory" (memory 6 1024 shared))
+  (func $__wasm_init_tls (export "__wasm_init_tls") (param i32)
+    (i32.const 232323)
+    drop
+  )
+  (func $__wbindgen_malloc (export "__wbindgen_malloc") (param i32) (result i32)
+    i32.const 999999
+  )
+  (func $start
+    i32.const 101010
+    drop
+  )
+  (func $__wbindgen_free (export "__wbindgen_free") (param i32 i32))
+  (global (export "__heap_base") i32 (i32.const 327680))
+  (global (export "__tls_size") i32 (i32.const 128))
+  (global (export "__tls_align") i32 (i32.const 4))
+  (global (export "__tls_base") (mut i32) (i32.const 0))
+  ;; stack pointer
+  (global (mut i32) (i32.const 65536))
+  (start $start)
+)
+
+(; CHECK-ALL:
+(module
+  (type (;0;) (func))
+  (type (;1;) (func (param i32)))
+  (type (;2;) (func (param i32) (result i32)))
+  (type (;3;) (func (param i32 i32)))
+  (import "env" "memory" (memory (;0;) 7 16384 shared))
+  (func (;0;) (type 0)
+    (local i32)
+    call $start
+    i32.const 327680
+    i32.const 1
+    i32.atomic.rmw.add
+    if  ;; label = @1
+      i32.const 393216
+      global.set 2
+      loop  ;; label = @2
+        i32.const 327684
+        i32.const 0
+        i32.const 1
+        i32.atomic.rmw.cmpxchg
+        if  ;; label = @3
+          i32.const 327684
+          i32.const 1
+          i64.const -1
+          memory.atomic.wait32
+          drop
+          br 1 (;@2;)
+        else
+        end
+      end
+      i32.const 1048576
+      call $__wbindgen_malloc
+      local.tee 0
+      i32.const 327684
+      i32.const 0
+      i32.atomic.store
+      i32.const 327684
+      i32.const 1
+      memory.atomic.notify
+      drop
+      block  ;; label = @2
+        local.get 0
+        br_if 0 (;@2;)
+        unreachable
+      end
+      global.set 3
+      global.get 3
+      i32.const 1048576
+      i32.add
+      global.set 2
+    else
+    end
+    i32.const 128
+    i32.const 4
+    drop
+    call $__wbindgen_malloc
+    global.set 1
+    global.get 1
+    call $__wasm_init_tls)
+  (func $__wbindgen_thread_destroy (type 0)
+    global.get 1
+    i32.const -2147483648
+    i32.ne
+    if  ;; label = @1
+      global.get 1
+      i32.const 128
+      call $__wbindgen_free
+    else
+    end
```

Comment:
This looks like it's trying to support multiple invocations of `__wbindgen_thread_destroy`, but I think we can probably skip that? In this destructor I think it's ok to assume it's the first and only time it's running (and basically have "UB" if it isn't)